.github/workflows/main.yml

```diff
@@ -23,6 +24,7 @@ jobs:
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
+        python -m pip install setuptools
```

Comment:
As the `setuptools` is a required package for insights-core to run on python 3.12, how about adding it to the `setup.py` directly, e.g. `'setuptools; python_version >= "3.12"'`?

Reply:
Good point.